Assets/Scripts/Army/Army.cs

```diff
@@ -92,26 +158,30 @@ private List<Vector2> IdentityComplex()
         var armyComplex = new List<Vector2>();
         foreach (var u in armyUnits)
         {
-            var identity = u.IdentityInfo;
-            var rgbIdentity = new Color(identity.r, identity.g, identity.b);
-            float h;
-            Color.RGBToHSV(rgbIdentity, out h, out _, out _);
-            //0->0, 1->2pi
-            var angle = 2 * Mathf.PI * h;
-            var complex = new Vector2(Mathf.Cos(angle), Mathf.Sin(angle));
-            armyComplex.Add(complex);
+            armyComplex.Add(
+                CalculateIdentityComplex(u)
+                );
         }
 
         return armyComplex;
     }
+
+    private Vector2 CalculateIdentityComplex(Unit u)
+    {
+        var identity = u.IdentityInfo;
+        var rgbIdentity = new Color(identity.r, identity.g, identity.b);
+        float h;
+        Color.RGBToHSV(rgbIdentity, out h, out _, out _);
```

Comment:
Consider making a utility function GetCalculationColorStruct in Army or ObjectIdentity that returns RGB/HSV/something else so there is one place that stores what color space we are using for colors